packages/react/react/src/hooks/useVisibilityEvent.spec.tsx

```diff
@@ -0,0 +1,41 @@
+import { fireEvent, render, renderHook } from '@testing-library/react';
+import { useState } from 'react';
+import { useVisibilityEvent } from './useVisibilityEvent';
+
+type VisibilityState = Document['visibilityState'];
+
+describe('useVisibilityEvent', () => {
+  it('When a user leaves the page.', () => {
+    const callbackMock = jest.fn();
+    renderHook(() => useVisibilityEvent(callbackMock));
+    expect(document.visibilityState).toBe('visible');
+
+    Object.defineProperty(document, 'visibilityState', { value: 'hidden' });
+    fireEvent(document, new Event('visibilitychange'));
+
+    expect(document.visibilityState).toBe('hidden');
```

Comment:
I think we're missing an additional `expect` that `callbackMock` is called here.